$NetBSD$

Adding NetBSD support. Setting CFLAGS, LDFLAGS for OpenGL, PNG, and SDL.

--- source/mupen64plus-video-rice/projects/unix/Makefile.orig	2012-03-10 18:31:45.000000000 +0000
+++ source/mupen64plus-video-rice/projects/unix/Makefile
@@ -60,6 +60,11 @@ ifneq ("$(filter GNU/kFreeBSD kfreebsd,$
   SO_EXTENSION = so
   SHARED = -shared
 endif
+ifeq ("$(UNAME)","NetBSD")
+  OS = NETBSD
+  SO_EXTENSION = so
+  SHARED = -shared
+endif
 ifeq ("$(OS)","NONE")
   $(error OS type "$(UNAME)" not supported.  Please file bug report at 'http://code.google.com/p/mupen64plus/issues')
 endif
@@ -104,9 +109,9 @@ endif
 
 # base CFLAGS, LDLIBS, and LDFLAGS
 OPTFLAGS ?= -O3
-CFLAGS += $(OPTFLAGS) -Wall -msse -ffast-math -fno-strict-aliasing -fvisibility=hidden -I../../src
+CFLAGS += $(OPTFLAGS) -Wall -msse -ffast-math -fno-strict-aliasing -fvisibility=hidden -I../../src $(shell cpuflags)
 CXXFLAGS += -fvisibility-inlines-hidden
-LDFLAGS += $(SHARED)
+LDFLAGS += -Wl,-rpath,${LIBDIR} $(shell pkg-config --libs libpng) $(SHARED)
 
 # Since we are building a shared library, we must compile with -fPIC on some architectures
 # On 32-bit x86 systems we do not want to use -fPIC because we don't have to and it has a big performance penalty on this arch
@@ -133,7 +138,7 @@ endif
 
 # set special flags per-system
 ifeq ($(OS), LINUX)
-  LDLIBS += -ldl
+  LDLIBS += ${DL_LIBS}
   # only export api symbols
   LDFLAGS += -Wl,-version-script,$(SRCDIR)/video_api_export.ver
 endif
@@ -142,14 +147,18 @@ ifeq ($(OS), OSX)
     ifeq ($(ARCH_DETECTED), 64BITS)
       CFLAGS += -pipe -arch x86_64 -mmacosx-version-min=10.5 -isysroot /Developer/SDKs/MacOSX10.5.sdk
       LDFLAGS += -bundle -framework OpenGL -arch x86_64
-      LDLIBS += -ldl
+      LDLIBS += ${DL_LIBS}
     else
       CFLAGS += -pipe -mmmx -msse -fomit-frame-pointer -arch i686 -mmacosx-version-min=10.5 -isysroot /Developer/SDKs/MacOSX10.5.sdk
       LDFLAGS += -bundle -framework OpenGL -arch i686
-      LDLIBS += -ldl
+      LDLIBS += ${DL_LIBS}
     endif
   endif
 endif
+ifeq ($(OS),NETBSD)
+  CFLAGS += $(shell pkg-config --cflags gl libpng)
+  LDFLAGS += $(shell pkg-config --libs gl libpng)
+endif
 
 # test for essential build dependencies
 ifeq ($(shell which pkg-config 2>/dev/null),)
@@ -182,6 +191,10 @@ ifeq ($(OS),LINUX)
     CFLAGS  += $(shell sdl-config --cflags)
     LDLIBS += $(shell sdl-config --libs)
 endif
+ifeq ($(OS),NETBSD)
+    CFLAGS  += $(shell sdl-config --cflags)
+    LDFLAGS += $(shell sdl-config --libs)
+endif
 
 # set mupen64plus core API header path
 ifneq ("$(APIDIR)","")
@@ -338,7 +351,7 @@ all: $(TARGET)
 
 install: $(TARGET)
 	$(INSTALL) -d "$(DESTDIR)$(PLUGINDIR)"
-	$(INSTALL) -m 0644 $(INSTALL_STRIP_FLAG) $(TARGET) "$(DESTDIR)$(PLUGINDIR)"
+	$(INSTALL) -m 0755 $(INSTALL_STRIP_FLAG) $(TARGET) "$(DESTDIR)$(PLUGINDIR)"
 	$(INSTALL) -d "$(DESTDIR)$(SHAREDIR)"
 	$(INSTALL) -m 0644 "../../data/RiceVideoLinux.ini" "$(DESTDIR)$(SHAREDIR)"
 
